src/go/rdctl/pkg/snapshot/manager_test.go

```diff
@@ -31,20 +31,53 @@ func TestManager(t *testing.T) {
 		paths, _ := populateFiles(t, true)
 		manager := newTestManager(paths)
 		invalidNames := []string{
-			"test!",
-			`"test"`,
-			`'test'`,
-			"testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest" +
-				"testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest",
-		}
-		for _, c := range strings.Split("!$^&*()[]{};:?/'` \\\"", "") {
-			invalidNames = append(invalidNames, fmt.Sprintf("invalid%sname", c)) // spellcheck-ignore-line
+			"", // empty string not allowed
+			" can't start with a space",
+			`can't end with a "space" `,
+			// 251 characters is too long (and the indentation here is what our linter demands)
+			"12345678911234567892123456789312345678941234567895123456789612345678971234567898" +
+				"12345678991234567890123456789112345678921234567893123456789412345678951234567896" +
+				"12345678971234567898123456789912345678901234567891123456789212345678931234567894" +
+				"12345678951",
 		}
 		for _, invalidName := range invalidNames {
 			if err := manager.ValidateName(invalidName); err == nil {
 				t.Errorf("name %q is invalid but no error was returned", invalidName)
 			}
 		}
+		err := manager.ValidateName(invalidNames[3])
+		if err == nil {
+			t.Error("oversize name is invalid but no error was returned")
```

Comment:
This was already checked in the loop above.

Also not a big fan of the hardcoded index `3` into the array. This is brittle because test cases may be added/inserted later.

Reply:
Easy way around it. I added a bit of code to verify that we're giving the correct error, not just any error.